api/turing/models/ensembling_job.go

```diff
@@ -85,6 +83,31 @@ func (r *InfraConfig) Scan(value interface{}) error {
 	return json.Unmarshal(b, &r)
 }
 
+// MarshalJSON marshals the Go struct into a JSON, most of the code here is duplicated
+// but requires some treatment to the overridden name
+func (r *InfraConfig) MarshalJSON() ([]byte, error) {
+	toSerialize := map[string]interface{}{}
+	toSerialize["service_account_name"] = r.ServiceAccountName
+
+	if r.ArtifactUri != nil {
+		toSerialize["artifact_uri"] = r.ArtifactUri
+	}
+
+	if r.EnsemblerName != nil {
+		toSerialize["ensembler_name"] = r.EnsemblerName
+	}
+
+	if r.Resources.IsSet() {
+		toSerialize["resources"] = r.Resources.Get()
+	}
+
+	if r.Env != nil {
+		toSerialize["env"] = r.Env
+	}
+
+	return json.Marshal(toSerialize)
+}
+
```

Comment:
This method ruins the entire idea (which is DRY and keeping a single source of truth) of using the generated code. 
Would this work?

```go
func (r *InfraConfig) MarshalJSON() ([]byte, error) {
         r.EnsemblerInfraConfig.SetServiceAccountName(r.ServiceAccountName)

	return json.Marshal(r.EnsemblerInfraConfig)
}
```

Reply:
I have to test this, there's some weird mechanic when you do `assert.Equal(t, expected, actual)` after querying from database. I'm very sure it will break all the tests because when you receive the payload from the API, you are setting the outer struct's ServiceAccountName. Because with this code you are setting the inner struct's ServiceAccountName while having the outer struct's ServiceAccountName. This actually sets 2 sources of truth.

Reply:
Yeah this wont work:
```
    ensembling_job_service_test.go:234:
                Error Trace:    ensembling_job_service_test.go:234
                                                        database.go:92
                                                        ensembling_job_service_test.go:203
                Error:          Not equal:
                                expected: &models.InfraConfig{EnsemblerInfraConfig:openapi.EnsemblerInfraConfig{ArtifactUri:(*string)(0xc0005125d0), EnsemblerName:(*string)(0x495ab50), ServiceAccountName:(*string)(0xc0005136f0), Resources:openapi.NullableEnsemblingResources{value:(*o
penapi.EnsemblingResources)(0xc0008724b0), isSet:true}, Env:(*[]openapi.EnvVar)(0xc00049e800)}, ServiceAccountName:"test-service-account-1"}
                                actual  : &models.InfraConfig{EnsemblerInfraConfig:openapi.EnsemblerInfraConfig{ArtifactUri:(*string)(0xc00007d5b0), EnsemblerName:(*string)(0xc00007d5c0), ServiceAccountName:(*string)(nil), Resources:openapi.NullableEnsemblingResources{value:(*openapi
.EnsemblingResources)(0xc00059aed0), isSet:true}, Env:(*[]openapi.EnvVar)(0xc000696d40)}, ServiceAccountName:"test-service-account-1"}
```